src/DetailsView/components/load-assessment-button.tsx

```diff
@@ -3,31 +3,104 @@
 
 import { AssessmentDataParser } from 'common/assessment-data-parser';
 import { InsightsCommandButton } from 'common/components/controls/insights-command-button';
+import { Tab } from 'common/itab';
+import { AssessmentStoreData } from 'common/types/store-data/assessment-result-data';
+import { TabStoreData } from 'common/types/store-data/tab-store-data';
+import { VersionedAssessmentData } from 'common/types/versioned-assessment-data';
+import { UrlParser } from 'common/url-parser';
 import { DetailsViewActionMessageCreator } from 'DetailsView/actions/details-view-action-message-creator';
+import { LoadAssessmentDialog } from 'DetailsView/components/load-assessment-dialog';
 import { LoadAssessmentHelper } from 'DetailsView/components/load-assessment-helper';
 import * as React from 'react';
 
 export type LoadAssessmentButtonDeps = {
     detailsViewActionMessageCreator: DetailsViewActionMessageCreator;
     assessmentDataParser: AssessmentDataParser;
+    urlParser: UrlParser;
     loadAssessmentHelper: LoadAssessmentHelper;
 };
 export interface LoadAssessmentButtonProps {
     deps: LoadAssessmentButtonDeps;
+    tabStoreData: TabStoreData;
+    assessmentStoreData: AssessmentStoreData;
+}
+export interface LoadAssessmentButtonState {
+    newTargetPageData: Tab;
+    prevTargetPageData: Tab;
+    loadedAssessmentData?: VersionedAssessmentData;
+    show: boolean;
 }
 
 export const loadAssessmentButtonAutomationId = 'load-assessment-button';
 
-export class LoadAssessmentButton extends React.Component<LoadAssessmentButtonProps> {
+export class LoadAssessmentButton extends React.Component<
+    LoadAssessmentButtonProps,
+    LoadAssessmentButtonState
+> {
+    public constructor(props) {
+        super(props);
+        this.state = {
+            loadedAssessmentData: null,
+            newTargetPageData: this.getNewTab(),
```

Comment:
Since we don't need this here or the child components, we can actually remove this entirely. The only place we would use it, line 87 in this file, can be replaced by using `this.props[...]`.

Reply:
added in logic that uses this. It should've been there in the first place, it was an oversight on my part.